### PR TITLE
Add isAvailable() function to RefToBaseProd

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

The `isAvailable()` member function is available in
* `edm::RefProd`
* `edm::Ref`
* `edm::Ptr`
* `edm::RefToBase`

but not in `edm::RefToBaseProd`, as was noticed in https://github.com/cms-sw/cmssw/issues/41843#issuecomment-1574078890. This PR adds the `isAvailable()` to `edm::RefToBaseProd`.

#### PR validation:

Unit tests pass

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Potentially to be backported to 13_0_X (and 13_1_X).

### CODE REVIEW

**Comment by makortel** [comment]
@Dr15Jones Can you think of any reason why not?

File: DataFormats/Common/test/reftobaseprod_t.cppunit.cc [hunk 1]
```diff
@@ -93,11 +93,13 @@ void testRefToBaseProd::constructTest() {
   CPPUNIT_ASSERT(!nulled);
   CPPUNIT_ASSERT(nulled.isNull());
   CPPUNIT_ASSERT(!nulled.isNonnull());
+  CPPUNIT_ASSERT(!nulled.isAvailable());
 
   RefToBaseProd<Dummy> nulledP;
   CPPUNIT_ASSERT(!nulledP);
   CPPUNIT_ASSERT(nulledP.isNull());
   CPPUNIT_ASSERT(!nulledP.isNonnull());
+  CPPUNIT_ASSERT(!nulled.isAvailable());
```
Comment (francescobrivio) [inline DataFormats/Common/test/reftobaseprod_t.cppunit.cc:102]: Shouldn't this be?
```suggestion
  CPPUNIT_ASSERT(!nulledP.isAvailable());
```

Comment (makortel) [reply DataFormats/Common/test/reftobaseprod_t.cppunit.cc:102]: Good catch, thanks!

**Comment by makortel** [comment]
(pushed the wrong branch, apparently was quick enough to fix to prevent everyone to be tagged)

File: DataFormats/Common/interface/RefToBaseProd.h [hunk 2]
```diff
@@ -66,6 +66,10 @@ namespace edm {
     /// Checks for non-null
     bool isNonnull() const { return product_.isNonnull(); }
 
+    /// Checks if collection is in memory or available
+    /// in the Event. No type checking is done.
+    bool isAvailable() const { return product_.isAvailable(); }
```
[the remark this answers is not in the document]
Comment (makortel) [reply DataFormats/Common/interface/RefToBaseProd.h:71]: Would it be good to add that note for the other Ref/Ptr/etc objects that have `isAvailable()` function with similar implementation?

[the remark this answers is not in the document]
Comment (Dr15Jones) [reply DataFormats/Common/interface/RefToBaseProd.h:71]: Probably?

[the remark this answers is not in the document]
Comment (makortel) [reply DataFormats/Common/interface/RefToBaseProd.h:73]: Added a comment

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+core

**Comment by rappoccio** [comment]
+1

**Pull request merged**